### PR TITLE
Update oauth2 gem dependency and replace rest-client with faraday

### PR DESCRIPTION
**Pull request by 97jaz**

### CODE REVIEW

File: lib/fhir_client/client.rb [hunk 1]
```diff
@@ -440,309 +442,42 @@ def scrubbed_response_headers(result)
...
-        headers = response.headers.each_with_object({}) { |(k, v), h| h[k.to_s.tr('_', '-')] = v.to_s; h }
-        res = {
-          code: response.code,
-          headers: scrubbed_response_headers(headers),
-          body: response.body
-        }
-
-        @reply = FHIR::ClientReply.new(response.request.args, res, self)
-      end
+      @reply = perform_request(method: :get, path: path, headers: headers)
```
Comment (97jaz) [inline lib/fhir_client/client.rb:445]: Each of the HTTP verbs has its own method in the old implementation. They contain a lot of duplicated and overly-complicated code. They don't all do exactly the same thing, so my refactoring here _does_ change behavior. That could be a problem, but it seems unlikely to me. The differences in the different methods seem arbitrary to me (but, of course, I could be wrong).

Comment (hernanat) [reply lib/fhir_client/client.rb:445]: I trust you 🖖 😉

File: lib/fhir_client/client.rb [hunk 2]
```diff
@@ -752,5 +487,42 @@ def build_url(path)
...
+      debug_body = (debug_body[0...limit] + "...") if debug_body.length > limit
+      FHIR.logger.debug("#{method_string} - Request: #{response.env.to_json}\nResponse:\nResponse Headers: #{res[:headers]}\nResponse Body: #{debug_body}")
+
+      FHIR::ClientReply.new(req, res, self)
+    rescue => e
+      # We're not telling Faraday to raise on HTTP errors, so if we're here,
+      # we didn't get a response from the server. We'll log and re-raise
+      # the error.
+      FHIR.logger.error "#{method_string} - Request: #{url} failed! No response from server: #{e}"
+      raise
```
Comment (hernanat) [inline lib/fhir_client/client.rb:525]: > We'll log and re-raise the error.

does just putting `raise` here implicitly re-raise the same error?

**Review by hernanat — APPROVED** [review]

File: lib/fhir_client/client.rb (hunk 1 above)
Comment (97jaz) [reply lib/fhir_client/client.rb:445]: Why, thank you, @hernanat

**Pull request merged**